Applications/Spire/Include/Spire/KeyBindings/KeyBindingsTableView.hpp

```diff
@@ -53,14 +63,40 @@ namespace Spire {
       */
       void set_width(int width);
 
+    protected:
+      bool eventFilter(QObject* watched, QEvent* event) override;
+      void hideEvent(QHideEvent* event) override;
+
     private:
       CustomGridTableView* m_table;
       QHeaderView* m_header;
+      bool m_can_delete_rows;
+      QVBoxLayout* m_delete_buttons_layout;
+      QWidget* m_delete_buttons_widget;
+      std::unordered_map<int, int> m_minimum_column_widths;
+      bool m_is_default_cell_selected;
+      std::set<Qt::Key> m_navigation_keys;
```

Comment:
Don't use `std::set` or `std::map` unless you make use of the sorted order.
Use hash tables instead. `QHash` is likely implemented for `Qt::Key`.

Reply:
Should I make it an unordered_set? I don't need a key/value pair, I only use this to find if a key is a member of that set.

Reply:
You can, `std::unordered_set` to` std::set` is what `std::unordered_map` is to `std::map`.
I am not sure whether the STL hasher is implemented for `Qt::Key`. You can use `QSet` though, which is the Qt implementation of `unordered_set`.